src/main/java/net/schmizz/sshj/xfer/scp/SCPEngine.java

```diff
@@ -39,17 +40,31 @@
         RECURSIVE('r'),
         VERBOSE('v'),
         PRESERVE_TIMES('p'),
-        QUIET('q');
+        QUIET('q'),
+        LIMIT('l', "");
 
         private final char a;
+        private String v;
```

Comment:
Ahh but this is even more evil, as this is an enum, setting the value for 1 process means that you're setting it (inadvertently) for all processes. Because there is only 1 instance of the same enum key in a JVM. So please refactor that and remove the `setValue(..)`